crates/services/executor/src/vm_pool.rs

```diff
@@ -0,0 +1,39 @@
+use std::cell::RefCell;
+
+use fuel_core_types::fuel_vm::interpreter::Memory;
+
+thread_local! {
+    static POOL: RefCell<Vec<Memory>> = RefCell::new(Vec::new());
+}
+
+/// Gets a new VM memory instance from the pool.
+pub(crate) fn get_vm_memory() -> Memory {
+    POOL.with(|pool| {
+        let mut pool = pool.borrow_mut();
+        pool.pop().unwrap_or_else(Memory::new)
```

Comment:
We should probably have some kind constraint on the max number of items allowed in each pool? Otherwise it will wait if there are none available.

Reply:
We didn't do this earlier, and I don't see why it would make sense to track the items here, instead of just limiting concurrency of the executor itself. That said, keeping a pool per thread is by itself problematic and not useful, as:

1. The threads are not CPU-pinned anyway
2. The async runtime might decide to create new threads arbitarily, which would essentially cause a memory leak here

So I' changed the code to not keep separate pools in 5caec42